Gravity/Gravity.Test.Integration/RSAPI_IntegrationTest.cs

```diff
@@ -17,388 +18,440 @@
+						Guid singleChoiceGuid = singleChoice.GetRelativityObjectAttributeGuidValue();
+						kCura.Relativity.Client.DTOs.Choice singleChoiceToAdd = new kCura.Relativity.Client.DTOs.Choice(singleChoiceGuid);
+						dto.Fields.Add(new FieldValue(testFieldGuid, singleChoiceToAdd));
+						break;
+					case RdoFieldType.SingleObject:
+						int objectToAttach =
+								_testObjectHelper.CreateTestObjectWithGravity<GravityLevel2>(sampleData as GravityLevel2);
+						dto.Fields.Add(new FieldValue(testFieldGuid, objectToAttach));
+						expectedData = (sampleData as GravityLevel2).Name;
+						break;
+					case RdoFieldType.MultipleObject:
+						IList<GravityLevel2> gravityLevel2s = (IList<GravityLevel2>)sampleData;
+						FieldValueList<kCura.Relativity.Client.DTOs.Artifact> objects = new FieldValueList<kCura.Relativity.Client.DTOs.Artifact>();
+						expectedData = new Dictionary<int, string>();
+						foreach (GravityLevel2 child in gravityLevel2s)
+						{
+							objectToAttachID =
+								_testObjectHelper.CreateTestObjectWithGravity<GravityLevel2>(child);
+							objects.Add(new kCura.Relativity.Client.DTOs.Artifact(objectToAttachID));
+							(expectedData as Dictionary<int, string>).Add(objectToAttachID, child.Name);
+						}
+						dto.Fields.Add(new FieldValue(testFieldGuid, objects));
+						break;
+					default:
+						dto.Fields.Add(new FieldValue(testFieldGuid, sampleData));
+						break;
+				}
+
+				WriteResultSet<RDO> writeResults = _client.Repositories.RDO.Create(dto);
+
+				if (writeResults.Success)
+				{
+					newArtifactId = writeResults.Results[0].Artifact.ArtifactID;
+					Console.WriteLine(string.Format("Object was created with Artifact ID {0}.", newArtifactId));
+				}
+				else
+				{
+					Console.WriteLine(string.Format("An error occurred creating object: {0}", writeResults.Message));
+					foreach (var result in writeResults.Results.Select((item, index) => new { rdoResult = item, itemNumber = index }).Where(x => x.rdoResult.Success.Equals(false))
+							.Where(y => y.rdoResult.Success.Equals(false)))
+					{
+						Console.WriteLine(String.Format("An error occurred in create request {0}: {1}", result.itemNumber, result.rdoResult.Message));
+					}
+				}
+
+				Console.WriteLine("Arrangement Complete....");
+
+				//Act
+				Console.WriteLine("Starting Act....");
+
+				object gravityFieldValue = null;
+
+				if (newArtifactId > 0)
+				{
+					GravityLevelOne testGravityObject = _testObjectHelper.ReturnTestObjectWithGravity<GravityLevelOne>(newArtifactId);
+					gravityFieldValue = testGravityObject.GetPropertyValue(objectPropertyName);
+					if (gravityFieldValue != null)
+					{
+						switch (fieldType)
+						{
+							case RdoFieldType.SingleObject:
+								gravityFieldValue = (gravityFieldValue as GravityLevel2).Name;
+								break;
+							case RdoFieldType.MultipleObject:
+								var resultsDictionary = new Dictionary<int, string>();
```

Comment:
Use [`ToDictionary`](https://www.dotnetperls.com/todictionary).

Reply:
you and Linq?  I gotta admit, it's starting to grow on me a bit.  Done.